### PR TITLE
Dilated/Atrous Conv implementation with cudnn

### PR DESCRIPTION
**Pull request by MlWoo**
As #24559 described, cudnn provided the same API for dilated/atrous Conv and non-dilated/atrous Conv.  It is more like LSTMBlockCell in tf.contrib.  Maybe I should apply for a new API  of that of Conv using cudnn.  I am not familiar with PR of tensorflow. I commented the old code other than delete it. Look forward to your suggestion.

### CODE REVIEW

**Comment by MlWoo** [comment]
I signed it

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @yifeif: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

File: tensorflow/python/ops/nn_ops.py [hunk 1]
```diff
@@ -944,6 +1113,15 @@ def __init__(self,
     self.data_format = data_format
     self.strides = strides
     self.name = name
+    self.conv_op = AtrousConvolution(
```
Comment (azaks2) [inline tensorflow/python/ops/nn_ops.py]: you need to change convolution itself not to use _WithSpaceToBatch

File: tensorflow/python/ops/nn_ops.py [hunk 2]
```diff
@@ -3413,6 +3592,7 @@ def conv1d(value,
            filters,
            stride,
            padding,
+           dilations,
```
Comment (azaks2) [inline tensorflow/python/ops/nn_ops.py]: there is already dilation param please resync

**Review by azaks2 — COMMENTED** [review]
Can we get rid of all non_atrous code?

**Comment by tamaranorman** [comment]
Currently the backwards kernel is not supported on CPU which is why the code looks like it does unfortunately. Note the code for the convolution method has been updated since this commit but it may be possible to do a device check running the original on CPU and an updated version on GPU.

**Comment by rthadur** [comment]
@MlWoo request to re base your branch

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @azaks2, @tamaranorman: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

**Comment by tamaranorman** [comment]
@MlWoo The code underneath has changed a lot so a rebase would be good so can see the changes on top of the current file. Also I am unable to see exactly what the gain is from doing this. Currently convolution supports dilation as a parameter but this code would not have a backwards pass which works on a CPU

**Comment by tensorflowbutler** [comment]
It has been 28 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by rthadur** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!

**Pull request closed without merging**